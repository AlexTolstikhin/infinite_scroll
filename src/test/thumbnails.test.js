import React from 'react';
import { shallow } from 'enzyme';
import Thumbsnails from '../components/thumbsnails';

describe('Thumbsnails', () => {
	it('should render loading', () => {
		const wrapper = shallow(<Thumbsnails />);
		wrapper.setState({
			loading: true
		});
		expect(wrapper.debug()).toMatchSnapshot()
	});
	it('should render No results', () => {
		const wrapper = shallow(<Thumbsnails />);
		wrapper.setState({
			fetchedTimestamps: []
		});
		expect(wrapper.debug()).toMatchSnapshot();
	});
	it('should render end of list', () => {
		const props = {
			lastTimeStamp: 1
		}
		const wrapper = shallow(<Thumbsnails {...props}/>);
		wrapper.setState({
			startingTimeStamp: 2,
			fetchedTimestamps: []
		});
		expect(wrapper.debug()).toMatchSnapshot();
	});
	it('should render 2 images', () => {
		const wrapper = shallow(<Thumbsnails />);
		wrapper.setState({
			fetchedTimestamps: ['foo', 'foo']
		});
		expect(wrapper.debug()).toMatchSnapshot();
	});
});
