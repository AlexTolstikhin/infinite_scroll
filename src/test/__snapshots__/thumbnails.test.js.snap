// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbsnails should render 2 images 1`] = `
"<div style={{...}}>
  <img alt=\\"foo\\" src=\\"http://hiring.verkada.com/thumbs/foo.jpg\\" style={{...}} />
  <img alt=\\"foo\\" src=\\"http://hiring.verkada.com/thumbs/foo.jpg\\" style={{...}} />
</div>"
`;

exports[`Thumbsnails should render No results 1`] = `
"<div style={{...}}>
  <h1 style={{...}}>
    No Results Found
  </h1>
</div>"
`;

exports[`Thumbsnails should render end of list 1`] = `
"<div style={{...}}>
  <h1 style={{...}}>
    No Results Found
  </h1>
  <h1 style={{...}}>
    End of List
  </h1>
</div>"
`;

exports[`Thumbsnails should render loading 1`] = `
"<h1 style={{...}}>
  $
  Loading...
</h1>"
`;
